meshmode/array_context.py

```diff
@@ -175,11 +180,22 @@ def __init__(self, array_context):
         self._array_context = array_context
 
 
-class ArrayContext:
+class CommonSubexpression(Tag):
```

Comment:
```suggestion
class CommonSubexpressionTag(Tag):
```
? Mostly to differentiate it a bit from the same in `pymbolic`, but I don't feel strongly about it.

Reply:
51d2f20